index.d.ts

```diff
@@ -50,6 +50,30 @@ declare namespace WebpackBuildNotifierPlugin {
      * Set to false to play no sound for compilation notifications. Takes precedence over the *sound* configuration option.
      */
     compilationSound?: string | false;
+    /**
+     * The default function to be run after the notify has fired
+     */
+    customCallback?: Function;
+    /**
+     * The default function to be run after the success notify has fired
+     * Set to false to fire no function for success notifications. Takes precedence over the *customCallback* configuration option.
+     */
+    successCustomCallback?: Function | false;
+    /**
+     * The default function to be run after the warning notify has fired
+     * Set to false to fire no function for warning notifications. Takes precedence over the *customCallback* configuration option.
+     */
+    warningCustomCallback?: Function | false;
+    /**
+     * The default function to be run after the failure notify has fired
+     * Set to false to fire no function for failure notifications. Takes precedence over the *customCallback* configuration option.
+     */
+    failureCustomCallback?: Function | false;
+    /**
+     * The default function to be run after the compilation notify has fired
+     * Set to false to fire no function for compilation notifications. Takes precedence over the *customCallback* configuration option.
+     */
+    compilationCustomCallback?: Function | false;
```

Comment:
Rather than adding a callback for each scenario, what would you think about consolidating to two callbacks: one for `onCompileStart`, and another which accepts a parameter to indicate the status of the completed build? Maybe something like this?

```
type CompilationStatus = 'success' | 'error' | 'warning';

onCompileStart?: (compilation: webpack.compilation.Compilation) => void;
onComplete?: (status: CompilationStatus, compilation: webpack.compilation.Compilation) => void;
```

I feel that the config is a bit bloated as it is, so this would help keep the surface area a bit more tidy. :)

Reply:
```updated the pr,
i'm not sure how type hinting goes down in JS though but i did it for TS)

removed the ability to pass false and it just calls the default OR the function provided and moved the functionality of CompilationStatus checking onto the implementation provided to the completeCallback.

Hopefully this is what you were after?

Thanks!```